### ISSUE TITLE
Yum_package constraints

### ISSUE THREAD

**Issue by chazzly** [chef/chef#3908]
I'm not sure if this is a bug or a clarification of intended behavior.

If package foo ver 1.2 is installed on a client and 1.3 is available on the yum server and I have a resource like:

```
yum_package 'foo >= 1.2'
```

Running chef-client 12.2.1, this would be considered "up to date" if either 1.2 or 1.3 was installed.

After updating to chef-client 12.4.1, the chef-run is upgrading to 1.3, even if 1.2 is installed.  I'm guessing this change came from PR #3155.  My question is - which is the DESIRED behavior, the old or the new?

I would think the old way since anything greater than OR EQUAL should be considered acceptable.

**Comment by lamont-granquist** [comment]
Yeah, having `:install` act like an `:upgrade` is wrong.  If the constraint is satisfied by the installed packages, nothing should be done.  Probably all its doing is checking the candidate_version against the installed_version when it needs to be checking the installed_version against the version constraint before doing the candidate_version check.

**Comment by chazzly** [comment]
I'm guessing this is the same root problem, but I could be wrong.  

I've recently run into a slightly different situation.  Again this was not a problem with chef-client 12.2.1, but becomes a problem in 12.4.1.  using the same resource definition:
<code>yum_package 'foo >= 1.2'</code>

where version 1.3 is installed and both 1.2 & 1.3 are available from the yum server.  
Ver 1.2 is identified as the "candidate" version and since I do not have allow_downgrade set to true, it errors saying the candidate verison is older ... even though 1.3 should meet the ">= 1.2" constraint.

**Comment by chazzly** [comment]
Forgot to include log sample
```shell
Chef::Exceptions::Package: yum_package[glibc >= 2.12-1.7.el6] (cars-oraclient::11g line 9) had an error: Chef::Exceptions::Package: Installed package glibc-2.12-1.149.el6_6.5 is newer than candidate package glibc-2.12-1.129.el6
```
Manual yum search:
````shell
[root@test]# yum list glibc --showduplicates

Installed Packages
glibc.x86_64                                                              2.12-1.149.el6_6.5                                                               @extras
Available Packages
glibc.i686                                                                2.12-1.129.el6                                                                   base
glibc.x86_64                                                              2.12-1.129.el6                                                                   base
glibc.i686                                                                2.12-1.149.el6_6.5                                                               extras
glibc.x86_64                                                              2.12-1.149.el6_6.5                                                               extras
```

**Comment by chazzly** [comment]
Any updates?

**Comment by lamont-granquist** [comment]
should be fixed by #6540
